python/lsst/cp/pipe/makeBrighterFatterKernel.py

```diff
@@ -0,0 +1,1380 @@
+      raw images, and performs the selected instrument signature removal tasks.
+      For the purpose of brighter-fatter coefficient calculation is it
+      essential that certain components of isr are *not* performed, and
+      recommended that certain others are. The task checks the selected isr
+      configuration before it is run, and if forbidden components have been
+      selected task will raise, and if recommended ones have not been selected,
+      warnings are logged.
+
+    - The gain of the each amplifier in the detector is calculated using
+      the photon transfer curve (PTC) method and used to correct the images
+      so that all calculations are done in units of electrons, and so that the
+      level across amplifier boundaries is continuous.
+      Outliers in the PTC are iteratively rejected
+      before fitting, with the nSigma rejection level set by
+      config.nSigmaClipRegression. Individual pixels are masked from
+      the image based on config.nSigmaClipGainCalc
+
+    - Each image is then cross-correlated with the one it's paired with
+      (with the pairing defined by the --visitPairs command line argument),
+      which is done either the whole-image to whole-image,
+      or amplifier-by-amplifier, depending on config.level.
+
+    - Once the cross-correlations have been calculated for each visit pair,
+      these are used to generate the correction kernel.
+      The maximum lag used, in pixels, and hence the size of the half-size
+      of the kernel generated, is given by config.maxLag,
+      i.e. a value of 10 will result in a kernel of size 2n-1 = 19x19 pixels.
+      Outlier values in these cross-correlations are rejected by using a
+      pixel-wise sigma-clipped thresholding to each cross-correlation in
+      the visitPair-length stack of cross-correlations.
+      The number of sigma clipped to is set by config.nSigmaClipKernelGen.
+
+    - Once DM-15277 has been completed, a method will exist to calculate the
+      empirical correction factor, config.biasCorr.
+      TODO: DM-15277 update this part of the docstring once the ticket is done.
+    """
+
+    RunnerClass = MakeBrighterFatterKernelTaskRunner
+    ConfigClass = MakeBrighterFatterKernelTaskConfig
+    _DefaultName = "makeBrighterFatterKernel"
+
+    def __init__(self, *args, **kwargs):
+        """Constructor for the MakeBrighterFatterKernelTask."""
+        pipeBase.CmdLineTask.__init__(self, *args, **kwargs)
+        self.makeSubtask("isr")
+
+        self.debug = lsstDebug.Info(__name__)
+        if self.debug.enabled:
+            self.log.info("Running with debug enabled...")
+            # if we're displaying, test it works and save the displays for later
+            # it's worth testing here as displays are flaky and sometimes
+            # can't be contacted, and given processing takes a while
+            # it's a shame to fail late due to display issues
```

Comment:
A very helpful comment. Please break into sentences with periods (I think the first line is one sentence
and the rest is another).